### PR TITLE
SnowflakeIO: do not quote empty fields 

### PR DESCRIPTION
**Pull request by turb**
When sending CSV to Snowflake for `COPY`, do not quote empty fields (`''`).

With some types, like integer, it leads to an error like this: `net.snowflake.client.jdbc.SnowflakeSQLException: Numeric value '' is not recognized`.

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

File: sdks/java/io/snowflake/src/main/java/org/apache/beam/sdk/io/snowflake/SnowflakeIO.java [hunk 1]
```diff
@@ -1192,7 +1192,13 @@ private String quoteField(String field) {
     }
 
     private String quoteField(String field, String quotation) {
```
Comment (Abacn) [inline sdks/java/io/snowflake/src/main/java/org/apache/beam/sdk/io/snowflake/SnowflakeIO.java:1194]: consider rename this private method (and its overload above) to sth like "quoteNonEmptyField" for clarity

**Review by Abacn — COMMENTED** [review]
Thanks for the fix. It sounds a valid edge case.

Ideally is there a way to unit test the change? e.g. to confirm

before: "1","","3","",
after "1",,"3",,

(not "1",,"3")

it does not introduce side effect like dropping a field, making the column mismatch

File: sdks/java/io/snowflake/src/main/java/org/apache/beam/sdk/io/snowflake/SnowflakeIO.java (hunk 1 above)
Comment (turb) [reply sdks/java/io/snowflake/src/main/java/org/apache/beam/sdk/io/snowflake/SnowflakeIO.java]: fixup 016e641396f4a5aad0d672a5d8e1de92a43eafa1

**Comment by turb** [comment]
> Ideally is there a way to unit test the change? e.g. to confirm

Added in fixup 54ec742d9dadc9900c63c00b78de68c9c63f68dc

**Review by Abacn — APPROVED** [review]
